lib/tapioca/helpers/config_helper.rb

```diff
@@ -60,9 +65,122 @@ def config_options(options)
         config = YAML.load_file(config_file, fallback: {})
       end
 
+      validate_config!(config_file, config)
+
       Thor::CoreExt::HashWithIndifferentAccess.new(config[command_name] || {})
     end
 
+    sig { params(config_file: String, config: T::Hash[T.untyped, T.untyped]).void }
+    def validate_config!(config_file, config)
+      # To ensure that this is not re-entered, we mark during validation
+      return if @validating_config
+      @validating_config = T.let(true, T.nilable(T::Boolean))
+
+      commands = T.cast(self, Thor).class.commands
+
+      errors = config.flat_map do |config_key, config_options|
+        command = commands[config_key.to_s]
+
+        unless command
+          next build_error("unknown key `#{config_key}`")
+        end
+
+        validate_config_options(command.options, config_key, config_options || {})
+      end.compact
+
+      unless errors.empty?
+        print_errors(config_file, errors)
+        exit(1)
+      end
+    ensure
+      @validating_config = false
```

Comment:
Do we want to set this to `false` here? Doesn't that make it so that invoking `config_options` will validate the options every time?

Reply:
Between different runs we should definitely re-validate configs. I guess what I'm not seeing is: in which scenario will `config_options` be invoked and `@validating_config` will be `true` so that validation is skipped? When do we actually skip the validation?

Reply:
The intention is to not reenter the method if `options` is called again while _inside_ the method. We want to clear that flag as we are exiting, it is not our concern if this method is called multiple times or not.

However, the original `options` method already does memoization on the returned result, so it will take care to not call this again afterwards.